lib/ansible/playbook/role/__init__.py

```diff
@@ -222,6 +222,10 @@ def _load_role_data(self, role_include, parent_role=None):
                 raise AnsibleParserError("The handlers/main.yml file for role '%s' must contain a list of tasks" % self._role_name,
                                          obj=handler_data, orig_exc=e)
 
+        directories = [directory for directory in self._loader.list_directory(self._role_path) if directory not in ('tasks', 'handlers', 'meta')]
```

Comment:
roles are still valid as 'vars' providers, or files or templates or just a dependency grouping, etc .. not sure this is a good idea

Reply:
roles can just provide 'content' , vars/defaults/files/templates for use in other things, so to be really empty it must have NOTHING, this only checks for a subset of that.